agent/otel.go

```diff
@@ -388,6 +392,16 @@ func (a *Agent) uploadTraces(ctx context.Context, taskDir string) error {
 	return catcher.Resolve()
 }
 
+func batchSpans(spans []*tracepb.ResourceSpans, batchSize int) [][]*tracepb.ResourceSpans {
+	// Batching algorithm from https://go.dev/wiki/SliceTricks#batching-with-minimal-allocation
```

Comment:
Nit: can you make this part of the function documentation and just explain why we do this?

Reply:
done